### PR TITLE
feat(Button): expose `ref` prop

### PR DESCRIPTION
**Pull request by dozortsev**
## Context

I need `ref` prop in `Button` as part of https://github.com/transferwise/neptune-web/pull/1399.

This change mainly because another today's change to `Select` component https://github.com/transferwise/neptune-web/pull/1445 ([see 👁️  this line](https://github.com/transferwise/neptune-web/pull/1445/files#diff-29d0efb3fed45db7c0cd30b2bd9190372a2db9f970e9ee367c106d230e115665R521)).

## Changes

- expose React `ref` prop in `Button`
- refactor `Button` to TypeScript

## Checklist

- [ ] Make PR title meaningful and follow [the commit lint format](https://github.com/transferwise/neptune-web/blob/master/CONTRIBUTING.md#versioning-and-commit-lint) (it will appear in the changelog)
- [ ] Changes are tested and all tests pass
- [ ] Changes meet [accessibility standards](https://github.com/transferwise/neptune-web/blob/main/ACCESSIBILITY.md) and there are no violations in the console
- [ ] Changes work in all supported browsers (don't forget IE11)
- [ ] You've updated the documentation if necessary

### CODE REVIEW

File: packages/components/src/actionButton/ActionButton.tsx [hunk 1]
```diff
@@ -1,12 +1,7 @@
...
-import { ButtonHTMLAttributes, MouseEvent, forwardRef } from 'react';
+import { ButtonHTMLAttributes, forwardRef } from 'react';
 
-type ActionButtonProps = {
-  className?: string;
-  disabled?: boolean;
-  text: string;
-  onClick: (event: MouseEvent<HTMLButtonElement>) => void;
-} & ButtonHTMLAttributes<HTMLButtonElement>;
+type ActionButtonProps = { text: string } & ButtonHTMLAttributes<HTMLButtonElement>;
```
Comment (dozortsev) [inline packages/components/src/actionButton/ActionButton.tsx:4]: all those props (`className`, `disabled` ...) are coming from `ButtonHTMLAttributes`

File: packages/components/src/button/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      className,
+      disabled,
+      htmlType = 'button',
+      loading = false,
+      priority = Priority.PRIMARY,
+      size = Size.MEDIUM,
+      type = ControlType.ACCENT,
+      ...rest
+    }: Props,
+    reference,
```
Comment (richardmccartney) [inline packages/components/src/button/Button.tsx:52]: Can we add a test just to ensure `reference` is working and is `true`

```js
  it(': set `reference` to be true on Button', () => {
    const reference = createRef<HTMLButtonElement>();
    const props = {
      reference,
    };

    expect(reference.current).toBeFalsy();

    render(<Button {...props}>Click me!</Button>);

    expect(reference.current).toBeTruthy();
  });
```

**Review by richardmccartney — CHANGES_REQUESTED** [review]
Let's just add a unit test to this component to test `ref` and we should be good 👍

File: packages/components/src/button/Button.tsx (hunk 2 above)
Comment (dozortsev) [reply packages/components/src/button/Button.tsx:52]: added the test https://github.com/transferwise/neptune-web/pull/1446/commits/b4e87a50708d306c415a86fb218b12515ad7a916

**Review by richardmccartney — APPROVED** [review]

**Pull request merged**